lib/app.dart

```diff
@@ -23,7 +31,45 @@ class App {
     _showContent(_currentPage);
     _renderProgress();
     container.classes.toggle("hidden", false);
-    interactive.Interactive(interactiveContainer);
+
+    initFirebase();
+  }
+
+  void initFirebase() async {
+    await fb.init();
+    fb.firebaseAuth.onAuthStateChanged.listen(_fbAuthChanged);
+    loginButton.onClick.listen((_) => fb.googleLogin());
+    logoutButton.onClick.listen((_) => fb.signout());
+  }
+
+  void _fbAuthChanged(firebase.User user) {
+    if (user == null) {
+      print("User signed out");
+      loginModal.removeAttribute("hidden");
+      _interactiveInstance?.clear();
+      return;
+    }
+
+    if (!fb_constants.allowedEmailDomains
+        .any((domain) => user.email.endsWith(domain))) {
+      print("error email domain not allowed");
+      loginError
+        ..removeAttribute("hidden")
+        ..innerText = "Email domain not allowed";
```

Comment:
Could you also add `firebase.auth().currentUser.delete()` here (or just `user.delete()` if it works) to delete the entry from firebase as I don't think we want to store the details of users that don't have access, and you could also potentially call `fb.signout()` if the delete doesn't sign the user out.

Reply:
Deleted - didn't know fb allows user to delete account from the client! Delete already signs out the user.